src/internal/m365/collection/drive/collections.go

```diff
@@ -772,14 +879,44 @@ func (c *Collections) PopulateDriveCollections(
 			c.CollectionMap[driveID] = map[string]*Collection{}
 			invalidPrevDelta = true
 
+			// Reset collections and stats counts since we're starting over.
 			c.resetStats()
+
+			stats = &driveEnumerationStats{}
 		}
 
 		for _, item := range page {
 			if el.Failure() != nil {
 				break
 			}
 
+			// Check if we got the max number of containers we're looking for and also
+			// processed items for the final container.
+			if limiter.enabled() {
+				if item.GetFolder() != nil || item.GetPackageEscaped() != nil {
+					// Don't check for containers we've already seen.
+					if _, ok := c.CollectionMap[driveID][ptr.Val(item.GetId())]; !ok {
+						cp, err := c.getCollectionPath(driveID, item)
+						if err != nil {
+							el.AddRecoverable(ctx, clues.Stack(err).
+								WithClues(ctx).
+								Label(fault.LabelForceNoBackupCreation))
+
+							continue
+						}
+
+						if cp.String() != lastFolderPath {
+							if limiter.atLimit(stats) {
+								break
+							}
+
+							lastFolderPath = cp.String()
+							stats.numContainers++
+						}
+					}
+				}
+			}
```

Comment:
I would like to see it there too, the problem is that it has to be able to break out of the for-loop iterating through the page of results. We could start adding more return values to `processItem`, but we're still going to have some form of information sharing and/or ownership issues

Reply:
I was thinking about that.  Instead of a return value, is there a way to pack this into `limiter.AtLimit` through mutations?  Eg:
```
for item in page {
    // expect mutations to limiter
    processItem(..., limiter) 

    // mutations from above get evaluated in this call
    // more or less the same thing if we also mutate/review stats as well.
    if limiter.atLimit() { break }
}
```

Reply:
yea, that could be possible. I'd say let's roll it into a future PR if we go that route